src/Microsoft.Health.Fhir.KeyVault/Registration/FhirServerBuilderSecretStoreRegistrationExtensions.cs

```diff
@@ -0,0 +1,53 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Microsoft.Azure.KeyVault;
+using Microsoft.Azure.Services.AppAuthentication;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Health.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.Core.Features.SecretStore;
+using Microsoft.Health.Fhir.Core.Registration;
+using Microsoft.Health.Fhir.KeyVault;
+using Microsoft.Health.Fhir.KeyVault.Configs;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class FhirServerBuilderSecretStoreRegistrationExtensions
+    {
+        private const string KeyVaultConfigurationName = "KeyVault";
+
+        public static IFhirServerBuilder AddSecretStore(this IFhirServerBuilder fhirServerBuilder, IConfiguration configuration)
+        {
+            EnsureArg.IsNotNull(fhirServerBuilder, nameof(fhirServerBuilder));
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+
+            var keyVaultConfig = new KeyVaultConfiguration();
+            configuration.GetSection(KeyVaultConfigurationName).Bind(keyVaultConfig);
```

Comment:
This whole section could be missing.

Reply:
Added some comments here. Also renamed `InMemorySecretStore` to `InMemoryKeyVaultSecretStore` and moved it to the Fhir.KeyVault project to make things clearer and avoid semantic confusion.